src/printer.js

```diff
@@ -562,7 +562,20 @@ function genericPrintNoParens(path, options, print) {
       parts.push("return");
 
       if (n.argument) {
-        parts.push(" ", path.call(print, "argument"));
+        if (n.argument.comments) {
+          parts.push(
+            n.loc.start.line === n.loc.end.line
```

Comment:
I think it happens to fix the original problem because the comment is forcing new lines (in the input) in this case; you can't repro the original problem with a single-line return.

However, this code is implying that we should always wrap the return in parens and put in on multiple lines if the original code is like that. We definitely want to avoid respecting the original formatting as much as possible. We want to format it consistently, and it also makes it harder for generated ASTs to use prettier.

I think this is a little too aggressive of a solution; we need to figure out something a little more scoped and either just move the comment up or do something specific to that scenario, and not change how we format return statements.